src/js/diff.js

```diff
@@ -960,7 +960,13 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
  *
  */
 fluid.diff.objectsEqual = function (leftObject, rightObject) {
-    if (typeof leftObject !== typeof rightObject) {
+    if (leftObject === null && rightObject === null) {
+        return true;
+    }
+    else if ((leftObject === null && rightObject !== null) || (leftObject !== null && rightObject === null)) {
```

Comment:
A way I can like to express this is 
if (leftObject === null ^ rightObject === null)

Reply:
Done.